Docs: Update descriptions of common options (#2019)
grafana/grafana-operator: docs/docs/overview.md

## ResyncPeriod

Grafana doesn't have any webhooks or similar ways of notifying the operator that a Grafana resource, like a dashboard, has changed.
Due to this the Grafana operator constantly polls the Grafana API to test for changes and overwrite the resources, reconciling towards the desired state.

We describe this loop as a synchronizing resources with Grafana instances.

To control how often this polling should occur, you can set the `spec.resyncPeriod` field.

> This field tells sets how often it should poll the Grafana instance for changes for the specific resource.

Reviewer: Two verbs instead of one ("tells sets"), I guess something like this would be fine: 

```suggestion
This field tells the operator how often it should poll the Grafana instance for changes for the specific resource.
```

> If a dashboard has changed, the operator will overwrite and synchronize the dashboard after `10m` per default.

Reviewer: ```suggestion
If a dashboard has changed, the operator will overwrite and synchronize the dashboard after `10m` by default.
```

> This can of course be annoying for developers actively updating a resource. The recommended workflow is to duplicate the dashboard/alert/other and work on the copy.

Reviewer: Should we mark the PR as fixing #1597 then?

Author: I think that line was present before my edits: https://github.com/grafana/grafana-operator/blob/129224eef2ae5ab09bb95a868a9cdc2e28013809/docs/docs/overview.md?plain=1#L35-L36

So it's not really fixed in this PR but rather in: https://github.com/grafana/grafana-operator/pull/1466

When finished, export the changes and update the resource manifest to update the original.

> This can be disabled by setting a value `0m`

Reviewer: ```suggestion
This can be disabled by setting the value to `0m`.
```

metadata:
  name: grafanadashboard-sample
spec:
  instanceSelector:
    matchLabels:
      dashboards: "grafana" # Notice the label
  json: ...
```

As it is a normal label selector, it is possible to leave it empty to match everything.

```yaml
apiVersion: grafana.integreatly.org/v1beta1
kind: GrafanaDashboard
metadata:
  name: match-all-in-namespace
spec:
  instanceSelector: {}
  json: ...
```

Additionally, the more powerful `matchExpressions` are available and can be used alone or in combination with `matchLabels`

```yaml
apiVersion: grafana.integreatly.org/v1beta1
kind: GrafanaDashboard
metadata:
  name: match-expressions
spec:
  instanceSelector:
    matchExpressions:
    - { key: dashboards, operator: In, values: ["grafana", "loki", "tempo"]}
    - { key: datasources, operator: NotIn, values: ["internal"]}
  json: ...
```

## AllowCrossNamespaceImport

Allows resources in one namespace to be applied to Grafana instances in other namespaces as well.

```yaml
apiVersion: grafana.integreatly.org/v1beta1
kind: GrafanaDashboard
metadata:
  name: match-across-namespaces
spec:
  allowCrossNamespaceImport: true
  instanceSelector: {}
  json: ...
```

> Disabled per default to avoid exposing possibly sensitive resources by accident.

Reviewer: ```suggestion
Disabled by default to avoid exposing possibly sensitive resources by accident.
```

Author: I've seen both used, but it looks like by default is the more correct way of phrasing it, I've changed it according to your suggestions :)

More information can be found in [#44](https://github.com/grafana-operator/grafana-operator-experimental/issues/44).

> `allowCrossNamespaceImport` is one-way mutable, you can always enable it(`true`) but never disable.

Reviewer: ```suggestion
`allowCrossNamespaceImport` is one-way mutable, you can always enable it (`true`) but never disable.
```
